src/otx/core/metrics/accuracy.py

```diff
@@ -1,16 +1,35 @@
-# Copyright (C) 2022 Intel Corporation
+# Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 #
-"""Module for defining hierarchical label accuracy metric."""
+"""Module for OTX accuracy metric used for classification tasks."""
 
 from __future__ import annotations
 
-from typing import Callable, Sequence
+from typing import TYPE_CHECKING, Callable, Literal, Sequence
 
 import torch
 from torch import nn
 from torchmetrics import Metric
-from torchmetrics.classification import Accuracy, MultilabelAccuracy
+from torchmetrics.classification.accuracy import Accuracy, MultilabelAccuracy
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+class CustomAccuracy(Metric):
+    """Skleton code for future implementation."""
+
+    def __init__(self, average: Literal["MICRO", "MACRO"]):
+        self.average = average
+
+    def update(self, preds: Tensor, target: Tensor) -> None:
+        """Update state with predictions and targets."""
+
+    def compute(self) -> Tensor:
+        """Compute the metric."""
+
+    def _compute_unnormalized_confusion_metrices_from_preds(self, preds: Tensor) -> None:
+        """Compute an (unnormalized) confusion matriix for every label group."""
```

Comment:
Why is this class necessary?

Reply:
This could be removed. It the skeleton code for the future work.

Reply:
Why not the future work just deriving from `Metric`?